tests/test_tiatoolbox.py

```diff
@@ -101,6 +103,93 @@ def _response_all_wsis(_response_ndpi, _response_svs, tmpdir_factory):
     return dir_path
 
 
+@pytest.fixture(scope="session")
+def _response_stainnorm_source(tmpdir_factory):
+    """
+    Sample pytest fixture for source image for stain normalisation
+    Download png image for pytest
+    """
+    source_file_path = tmpdir_factory.mktemp("data").join("source.png")
+    if not os.path.isfile(source_file_path):
+        print("Downloading Source Image for stain normalisation")
+        r = requests.get(
+            "https://warwick.ac.uk/fac/sci/dcs/research/tia/tiatoolbox"
+            "/files/source.png"
+        )
+        with open(source_file_path, "wb") as f:
+            f.write(r.content)
+
+    else:
+        print("Skipping Source Image")
+
+    return source_file_path
+
+
+@pytest.fixture(scope="session")
+def _response_stainnorm_target(tmpdir_factory):
+    """
+    Sample pytest fixture for target image for stain normalisation
+    Download png image for pytest
+    """
+    target_file_path = tmpdir_factory.mktemp("data").join("target.png")
+    if not os.path.isfile(target_file_path):
+        print("Downloading Target Image for stain normalisation")
+        r = requests.get(
+            "https://warwick.ac.uk/fac/sci/dcs/research/tia/tiatoolbox"
+            "/files/target.png"
+        )
+        with open(target_file_path, "wb") as f:
+            f.write(r.content)
+
+    else:
+        print("Skipping Target Image")
+
+    return target_file_path
```

Comment:
Since the target image is small, should be not included it as a recourse in the pip package so that it is included with the python code. Just seems a bit silly to download it each time. Especially as it will be used outside of unit tests also.

Reply:
It's up to us I guess. I did it the way you are describing initially and then changed to how it is structured now. Assume that the user will want to supply their own target image? Happy to add an example target image instead within the repo. What do we think?